TwoPaneLayout/README.md

```diff
@@ -1,16 +1,34 @@
+When the app is spanned across a vertical hinge or fold, or when the width is larger than height of the screen on a large-screen device, pane 1 will be placed on the left, while pane 2 will be on the right. If the device rotates, the app is spanned across a horizontal hinge or fold, or the width is smaller than the height of screen on large-screen device, pane 1 will be placed on the top and pane 2 will be on the bottom.
 
-The TwoPaneLayout is able to assign children widths or heights according to their weights provided using the TwoPaneScope.weight modifier.
+## Weight modifier
 
-- For large screens, if no weight is provided, the two panes will be displayed equally. If weight is provided, then the layout will be split up proportionally according to the weight.
-- For foldables with a separating folding feature, the two panes will always be split up according to the folding feature boundaries (with or without weight). A folding feature is considered separating when a foldable device is half-opened or has a hinge that physically divides the screens. If a foldable device does not have a separating folding feature, then it will either be treated as a large screen or a single screen depending on its size and the provided weight.
-- If running on regular single-screen device, there will be only one pane visible. The other pane will be overlaid and navigation will be available to switch between two panes, regardless the weight.
+TwoPaneLayout is able to assign children widths or heights according to their weights provided using the `TwoPaneScope.weight` modifier. This only effects the layout for large screen and foldable devices, but for single-screen devices, there will still only be one pane visible, regardless of the weight.
+
+For large screens:
+
+- No weight &rarr; two panes displayed equally
+- Weight &rarr; layout split up proportionally according to the weight
+
+For foldables:
+
+- Separating fold &rarr; layout split up according to fold boundaries (with or without weight)
+- Non-separating fold &rarr; device treated as a large screen or single-screen depending on its size
+
+## Navigation support
+
+TwoPaneLayout offers two constructors with customizable navigation support.
+
+The first is the original `TwoPaneLayout` constructor with an extra `NavHostController` parameter. This gives you the option to create and control your own navController, which will still be used to navigate between panes 1 and 2 in single pane mode. The content passed into this constructor should be under the `TwoPaneScope`.
+
+The second is the `TwoPaneLayoutNav` constructor, which accepts an array of destinations, instead of only two panes of content. This highly customizable constructor can be used in cases where an app has more than two app-level destinations, and provides more flexibility with navigation options and the navigation flow. The content passed into this constructor should be under the `TwoPaneNavScope`.
+
+## API reference
```

Comment:
For `API`, can we please separate them into the related session? Start with `TwoPaneLayout`, then `PaneMode`, then `weight`, then `TwoPaneLayoutNav`?

Reply:
updated the readme into new sections, let me know if you think it looks better😄